### PR TITLE
feat: Button support lineHeight token

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/46883

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Button support `contentLineHeight` component tokens.        |
| 🇨🇳 Chinese |       Button 组件支持配置 `contentLineHeight` 系列组件 token。    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/theme/themes/shared/genFontSizes.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 // https://zhuanlan.zhihu.com/p/32746810
+export function getLineHeight(fontSize: number) {
+  return (fontSize + 8) / fontSize;
```
Comment (crazyair) [inline components/theme/themes/shared/genFontSizes.ts:3]: 不用以前的方法，现在直接用 加 8 除以 fontSize 了？

File: components/theme/themes/shared/genFontSizes.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
 // https://zhuanlan.zhihu.com/p/32746810
```
Comment (crazyair) [inline components/theme/themes/shared/genFontSizes.ts:1]: 这个应该放下面那个方法上

File: components/theme/themes/shared/genFontSizes.ts (hunk 1 above)
Comment (MadCcc) [reply components/theme/themes/shared/genFontSizes.ts:3]: 以前也是这样的啊

Comment (crazyair) [reply components/theme/themes/shared/genFontSizes.ts:3]: 看到 20 行也改用这个抽离的方法了，没问题了

**Review by li-jia-nan — APPROVED** [review]

**Review by zombieJ — APPROVED** [review]

**Review by kiner-tang — APPROVED** [review]

**Review by crazyair — APPROVED** [review]
image
没问题了

**Comment by li-jia-nan** [comment]
是不是还要再往 master merge 一下？

**Pull request merged**